src/vehicle_gui.cpp

```diff
@@ -1373,22 +1365,22 @@ uint ShowRefitOptionsList(int left, int right, int y, EngineID engine)
 	/* Draw nothing if the engine is not refittable */
 	if (HasAtMostOneBit(cmask)) return y;
 
+	std::string str;
 	if (cmask == lmask) {
 		/* Engine can be refitted to all types in this climate */
-		SetDParam(0, STR_PURCHASE_INFO_ALL_TYPES);
+		str = GetString(STR_PURCHASE_INFO_REFITTABLE_TO, STR_PURCHASE_INFO_ALL_TYPES, cmask);
```

Comment:
```suggestion
		str = GetString(STR_PURCHASE_INFO_REFITTABLE_TO, STR_PURCHASE_INFO_ALL_TYPES, std::monostate{});
```
As far as I can determine `cmask` isn't used, though something is needed because `...REFITTABLE_TO` has a `{STRING2}` and as such requires two parameters.

I have primarily written this because I noticed that `cmask` was not set in this branch before and started wondering why. It's not that passing `cmask` is conceptually bad, it's bad when looking at it from a purist's point of view. With the addition of `cmask`, the parameters to all these calls are the same except for the second one. Maybe put that in `str` and have one `GetString(STR_PURCHASE_INFO_REFITTABLE_TO, str, cmask)` in the `DrawStringMultiLine`?

In any case, feel free to keep it this way.

Reply:
I did have it this way initially and then changed it for some reason. Going mad I suspect.